### PR TITLE
ROX-9739: Add global scope check

### PR DESCRIPTION
**Pull request by janisz**
## Description

Add scope checker to store just like we have now in datastore level.
To do that we need to add missing resources to resources list and relax validation on roxvet.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI

### CODE REVIEW

File: tools/generate-helpers/pg-table-bindings/permission_checker.go.tpl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    UpsertManyAllowed(ctx context.Context, keys ...sac.ScopeKey) (bool, error)
+    DeleteAllowed(ctx context.Context, keys ...sac.ScopeKey) (bool, error)
+{{- end }}
+
+{{- if $singlePK }}
+    GetIDsAllowed(ctx context.Context) (bool, error)
+    GetManyAllowed(ctx context.Context) (bool, error)
+    DeleteManyAllowed(ctx context.Context, keys ...sac.ScopeKey) (bool, error)
+{{- end }}
+
```
[the remark this answers is not in the document]
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/permission_checker.go.tpl:32]: I removed it on @connorgorman request https://github.com/stackrox/stackrox/pull/961#discussion_r831414313

**Comment by janisz** [comment]
@rhybrillou PTAL

File: tools/generate-helpers/pg-table-bindings/permission_checker.go.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhybrillou) [reply tools/generate-helpers/pg-table-bindings/permission_checker.go.tpl]: I think there was a misunderstanding here. @connorgorman can probably clarify again.

What I understood was that the `GetKeysToIndex` and `AckKeysToIndex` (useful in the key-value + bleve datastore architecture) are becoming obsolete with Postgres, which indexes the data on create/update/delete operations (hence Noop function bodies to keep the store interface compatibility).
I think `Walk` still is a valid datastore operation, which either requires full resource access (pattern "apply function on all items read from the storage"), or has to perform actions only on objects which are in scope (in which case the database has to apply scope filters).

File: tools/generate-helpers/pg-table-bindings/store_test.go.tpl [hunk 2]
```diff
@@ -68,6 +74,10 @@ func (s *{{$namePrefix}}StoreSuite) TestStore() {
 	s.Nil(found{{.TrimmedType|upperCamelCase}})
 
     {{if not .JoinTable -}}
+    {{- if or (.PermissionChecker) (.Type | isGloballyScoped) }}
+    withNoAccess := sac.WithNoAccess(ctx)
```
Comment (rhybrillou) [inline tools/generate-helpers/pg-table-bindings/store_test.go.tpl:78]: nit: naming -> withNoAccessCtx (let the reader know the variable is a context)

**Review by rhybrillou — APPROVED** [review]

File: tools/generate-helpers/pg-table-bindings/permission_checker.go.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (connorgorman) [reply tools/generate-helpers/pg-table-bindings/permission_checker.go.tpl]: Yeah anything about indexing can be removed here. Walk is still a valid datastore operation

**Pull request merged**